pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+type Endpoint struct {
+	// Required: The IP of this endpoint.
+	//FIXME: should we allow hostname-or-IP?  If so, maybe in other places.
+	IP string
+
+	// The ports exposed on this endpoint.
+	// FIXME: this would be nicer as a map of name -> struct
```

Comment:
I know we have decided twice NOT to do maps, but It was annoying as I wrote it up, so I wanted to put it on the table before someone asked.

I propose to remove this FIXME and that we re-evaluate this decision post 1.0